### ISSUE TITLE
InputNumber to support allowClear

### ISSUE THREAD

**Issue by janarvaezkng** [ant-design/ant-design#50885]
### What problem does this feature solve?

At times in a Form, having an undefined value has a specific meaning (like clearing this value from the DB on submit), so being able to clear the value from an InputNumber the same way an Input does would be great.

### What does the proposed API look like?

<InputNumber allowClear />, same as <Input allowClear />

**Comment by afc163** [comment]
See https://github.com/ant-design/ant-design/issues/18566#issuecomment-527832096

**Comment by MadCcc** [comment]
Since InputNumber has supported `suffix`, it could be implemented easily.

**Comment by MadCcc** [comment]
But there's still one question, which value should it be after clearing, 0 or undefined?

**Comment by janarvaezkng** [comment]
> See https://github.com/ant-design/ant-design/issues/18566#issuecomment-527832096

I guess it could just be floating to the left of these arrows? I was about to say "the same way as in the Select" but then realised the Select overlaps the arrow and it's still usable because you can click anywhere 😬 

> But there's still one question, which value should it be after clearing, 0 or undefined?

I would say `undefined`, as `0` is still a value in most cases (or it could be configurable? `allowClear={{ clearValue: 0 }}`)

**Comment by janarvaezkng** [comment]
Just commenting so it doesn't become inactive and then auto closed...

**Comment by zombieJ** [comment]
> final conclusion?

Maybe add clear button before the change trigger button?

**Comment by janarvaezkng** [comment]
Any updates?

**Comment by Linkodt** [comment]
We can have a design where using allowClear requires setting a default number. For instance, with `<InputNumber allowClear={8} /> `clicking the clear icon will revert the value to 8. The defaultValue would be 0 by default.

This is my suggestion.

if it is approved,I want to resolve it by myself. This is my first time to contribute, if something wrong, please point out

**Comment by Wxh16144** [comment]
`<InputNumber allowClear={8} />` This code smells bad, it should be a boolean. Your approach is no different from controlled setting values.

**Comment by Linkodt** [comment]
> `<InputNumber allowClear={8} />` This code smells bad, it should be a boolean. Your approach is no different from controlled setting values.

yes, you are right. So the type of allowClear is boolean. But return to the top question what is the number after clear.  Maybe we cloud add another prop to config it?

thx for ur reply
